constraint/pkg/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,24 @@
+// package instrumentation defines primitives to
+// support more observability throughout the contraint framework.
+package instrumentation
+
+// NamedStats covers primitive for many heterogenoues statstics and metrics
+// to be represented to consumers.
+type NamedStats struct {
+	StatsName string
+	Stats     []*Stat
```

Comment:
What is an example use case that would leverage a list of stats here?

Reply:
the current use case ( `evaluation metadata` / `perf metrics` showcase this:

-we have the name of all the stats gathered under `EvaluationMetadata`; `Name` here is the name of the collection of `Stat` structs
-two `Stat`s structs:
--`templateRunTimeMillies` & `engineType` each recording their own `Value` and `Description` & `Name`; `Name` here is the name of the `Stat` 

does this help? If so, I can add the explanation in as comments.

Reply:
Why would the collection of stat structs have a name? What would be an example name?

Reply:
We removed `NamedStats`